app/joinCircle.js

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useEffect } from "react";
+import {
+    SafeAreaView,
+    Text,
+    View,
+    TextInput,
+    TouchableOpacity,
+    Image
+} from "react-native";
+import { styled } from "nativewind";
+import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
+import { Button } from "../components/Buttons";
+import { BarCodeScanner } from "expo-barcode-scanner";
+
+const StyledSafeArea = styled(SafeAreaView);
+const StyledView = styled(View);
+const StyledText = styled(Text);
+const StyledInput = styled(TextInput);
+
+export default function Page() {
+    const [code, setCode] = useState("");
+    const [hasPermission, setHasPermission] = useState(null);
+    const [scanned, setScanned] = useState(false);
+
+    useEffect(() => {
+        const getBarCodeScannerPermissions = async () => {
+            const { status } = await BarCodeScanner.requestPermissionsAsync();
+            setHasPermission(status === "granted");
+        };
+
+        getBarCodeScannerPermissions();
+    }, []);
+
+    const handleBarCodeScanned = ({ type, data }) => {
+        setScanned(true);
+        if (type == "org.iso.QRCode") {
+            alert(
+                `Bar code with type ${type} and data ${data} has been scanned!`
+            );
+        } else {
+            alert("Invalid code type!");
+        }
+    };
+
+    if (hasPermission === null) {
+        return <Text>Requesting for camera permission</Text>;
+    }
+    if (hasPermission === false) {
+        return <Text>No access to camera</Text>;
+    }
+
+    return (
+        <StyledSafeArea className="bg-offblack border" style={{ flex: 1 }}>
+            <KeyboardAwareScrollView>
```

Comment:
add `bounces={false}` here to remove make the page less jumpy with the scrolling